src/jit/simdcodegenxarch.cpp

```diff
@@ -1199,15 +1199,15 @@ void CodeGen::genSIMDIntrinsicRelOp(GenTreeSIMD* simdNode)
 
                 inst_RV_RV(INS_pmovmskb, intReg, tmpReg1, simdType, emitActualTypeSize(simdType));
                 // There's no pmovmskw/pmovmskd/pmovmskq but they're not needed anyway. Vector compare
-                // instructions produce "all ones"/"all zeroes" components and pmovmskb extracts a 
-                // subset of each component's ones/zeroes. In the end we need to know if the result is 
-                // "all ones" where the number of ones is given by the vector byte size, not by the 
-                // vector component count. So, for AVX we need to compare to 0xFFFFFFFF and for SSE we 
-                // need to compare to 0x0000FFFF. 
-                // Note that -1 is used instead of 0xFFFFFFFF, on x64 emit doesn't correctly recognize 
-                // that 0xFFFFFFFF can be encoded in a single byte and emits the longer 3DFFFFFFFF 
+                // instructions produce "all ones"/"all zeroes" components and pmovmskb extracts a
+                // subset of each component's ones/zeroes. In the end we need to know if the result is
+                // "all ones" where the number of ones is given by the vector byte size, not by the
+                // vector component count. So, for AVX registers we need to compare to 0xFFFFFFFF and
+                // for SSE registers we need to compare to 0x0000FFFF.
+                // Note that -1 is used instead of 0xFFFFFFFF, on x64 emit doesn't correctly recognize
+                // that 0xFFFFFFFF can be encoded in a single byte and emits the longer 3DFFFFFFFF
                 // encoding instead of 83F8FF.
-                getEmitter()->emitIns_R_I(INS_cmp, EA_4BYTE, intReg, compiler->canUseAVX() ? -1 : 0x0000FFFF);
+                getEmitter()->emitIns_R_I(INS_cmp, EA_4BYTE, intReg, emitActualTypeSize(simdType) == 32 ? -1 : 0x0000FFFF);
```

Comment:
I hope I got this right, if actual size is 32 then it uses YMM registers. Otherwise it uses XMM registers, even when actual size is 8 (in the case of `Vector2`).